src/com/owncloud/android/ui/activity/ReceiveExternalFilesActivity.java

```diff
@@ -398,11 +404,15 @@ private void populateDirectoryList() {
         if (mFile != null) {
             // TODO Enable when "On Device" is recovered ?
             Vector<OCFile> files = getStorageManager().getFolderContent(mFile/*, false*/);
-            List<HashMap<String, OCFile>> data = new LinkedList<>();
+            files = sortFileList(files);
+
+            List<HashMap<String, Object>> data = new LinkedList<>();
             for (OCFile f : files) {
-                HashMap<String, OCFile> h = new HashMap<>();
+                if (f.isFolder()) {
```

Comment:
Hmm, that is exactly why I put it there. since I upload a file (or multiple) I should only be able to choose a folder, right? So wouldn't that not be a bug on master? If I am wrong I'll remove it, takes just a second :)